pyblish_rpc/service.py

```diff
@@ -124,6 +124,26 @@ def _plugin_from_name(self, name):
         plugins = pyblish.lib.ItemList("__name__", self._plugins)
         return plugins[name]
 
+    def emit(self, signal, kwargs):
+        """Parse the serialised kwargs into python classes, ei. instances and
+        plugins.
+        """
+
+        if signal == "instanceToggled":
+            for instance in self._context:
+                if instance.data["name"] == kwargs["instance"]["json"]["name"]:
+                    kwargs["instance"] = instance
+                    break
+
+        if signal == "pluginToggled":
+            for plugin in self._plugins:
+                if plugin.id == kwargs["plugin"]["json"]["id"]:
```

Comment:
It still looks like `kwargs` is a lot more than what is necessary. :S

All you need to determine the instance is the name, I would only transmit the name. There is a performance penalty for data sent as well, the less the better.

```python
# Something like this?
emit("instanceToggled", instance="myInstance", new_value=value, old_value=not value)
```

There is one more thing about this approach which isn't waterproof.

If the emitter isn't including `instance` in `kwargs`, `instance` will still get added to the final output. This is fine at the moment, even though it is very hard-coded and tight coupling between the emitter and this particular function, but it may happen that we emit this or other signals from multiple places and that one of those places doesn't include it, causing hard-to-find bugs, as this would accept it anyway.

So I would actually go with..

```python
if "instance" in kwargs:
  # convert instance
```

..instead. What do you think?